chain/network/src/peer/codec.rs

```diff
@@ -90,6 +91,10 @@ impl Decoder for Codec {
         } else {
             let res = Some(Ok(buf[4..4 + len as usize].to_vec()));
             buf.advance(4 + len as usize);
+            // Fix memory leak see (#TODO)
+            if buf.is_empty() && buf.is_empty().not() {
```

Comment:
```suggestion
            if buf.is_empty() && !buf.is_empty() {
```

One *can* `.not()`, but this isn't remotely idiomatic. 

Also -- this is `false`, no?

Reply:
@matklad I updated the logic. This isn't really ready for review, hence the "draft".